src/components/Theme/src/Theme.vue

```diff
@@ -20,21 +21,66 @@ function isVueComponent(value) {
 	return value && value.render;
 }
 
+// returns true if value is an array
+// of objects with id fields
+function isKeyedArray(value) {
```

Comment:
Should we just name these functions `isProfilesArray`, `profilesArrayIntoObject`, etc? Since this fixes an issue with the `theme.profiles` property and we don't have any other arrays in the theme data?

Unless we expect to add more arrays to the theme data. But if the idea is to eventually change `theme.profiles` to an object in the future, making it clear these functions are specifically to convert the profiles array might make cleanup easier down the line.

Reply:
Had a sync with Bashu the other day and he showed me some theme data and they're using keyed arrays for the secondary colors also... e.g.,

```
secondaryColors: [
  { id: '--whatever', color: 'whatever' },
  { id: --something-else', color: 'something-else' },
]
```

I think we should leave the name and behavior as-is, since it solves the problem in general and isn't overfitted specifically for profiles.

Reply:
I added a comment above the keyedArray function to state that it's primarily used for `profiles`